### DISCUSSION TITLE
create a new map

### DISCUSSION THREAD

**Discussion by xia0long** [carla-simulator/carla#4245] [General]
Hi there!

I plan to do some experiments about reinforcement learning in Carla, As the maps offered in Carla are too complicated so I decide to create a map myself. As I know the Roadrunner(officially recommended) is a paid software, I'm curious about is there any other alternative software to create a new map? 

Any advice is much appreciated.

**Comment by apiszcz** [comment]
CARLA talks 2020 show Autodesk, RoadRunnder and Blender as options.

**Reply by Zeinab-Rahmani** [reply]
Hello. I also wanted to use a simple map and I think you can use this link [OpenStreetMap](https://carla.readthedocs.io/en/latest/tuto_G_openstreetmap/).

**Comment by ZXJohn** [comment]
hello,i'm  encountering  the same problem. And the softewares mentioned above can only output the file of .fbx, while ingesting maps need .xodr and pared .fbx. So i tried Train3Dbuilder,but the result is far from fine still. At last,have you find any alternative？I will be very appreciated if you could reply me something .

**Comment by SExpert12** [comment]
RoadRunner trial version is available.

**Reply by tarynnielsen** [reply]
I've reached out to ask for the trial version multiple times. How did you get in contact with them?

**Reply by SExpert12** [reply]
If you are a student then you can contact the person who is managing the software purchase in your institute. Or on mathwork site request for trial version. Hope this would help.

**Reply by ADASDevelopment** [reply]
> I've reached out to ask for the trial version multiple times. How did you get in contact with them?

We tried to reach them a lot, they just didn't give a demo directly and we reached the demo by going through the purchasing process. Then they sent a demo and after it was completed, we made the purchase.

**Comment by SExpert12** [comment]
Best of luck.
Let me know if you need any help in RoadRunner map creation or any importing issues in CARLA. Importing RoadRunner map to CARLA is also complicated at times.

**Reply by 15259400687** [reply]
I hava some problems in making custom map with roadrunner.First,why is it so dark when I export a map with roadrunner and run it in CarlaUE? I tried to set BP_Sky like other maps, but the problem still occurs. Secondly, I tried to run Carla's related scripts in a custom map, but the problem of vehicles flying directly into the sky occurs, which is very different from the map that comes with Carla. I don't know if you have encountered these problems. I wish you a happy life here. Thanks.

**Comment by SExpert12** [comment]
Should I close this issue now?

**Comment by SExpert12** [comment]
Can you tell me the steps you follow to import the road runner map?

**Reply by 15259400687** [reply]
I follow the link to import my roadrunner  map. https://carla.readthedocs.io/en/0.9.10/tuto_A_add_map/#deprecated-ways-to-import-a-map

**Comment by SExpert12** [comment]
You need to go to the mathworks site and follow the instruction given by them. You need to import all the plugins to the specify folder. Follow it and tell me if you have any problem still.
https://in.mathworks.com/help/roadrunner/ug/export-to-carla.html
Hope this helps

**Reply by 15259400687** [reply]
Thank you for your suggestion. I tried to operate on the link you provided. But I encountered an error while importing the Roadrunner plugin. My system is Windows 10, and the Carla version is 0.9.10. After placing the Roadrunner plugin in the CarlaUE4 \ Unreal file, right-click on the. uproject file and select Generate Visual Studio project files. Subsequently, an error occurred during the make launch: E:/carla0.9.10/carla/Unreal/carlaUE4/Plugins/LoadRunerCarlaDatasmith/Source/LoadRunerCarlaDatasmith/Private/LoadRunerCarlaDatasmith BPLibrary.cpp (96): error C2039: "GetPackage": Not a member of "UStaticMesh"
C: \ Program Files \ Epic Games \ UE-4.24 \ Engine \ Source \ Runtime \ Engine \ Classes \ Animation/MurphTarget. h (15): Note: Refer to the declaration of "UStaticMesh"
Make: * * * [CarlaUE4Editor] Error 5
May I ask if there is any way to solve this problem? Thank you very much for your reply.
I wish you all the best here.

**Comment by SExpert12** [comment]
Can you send me the snapshot of all the plug-ins. Actually you need to copy all the plug-ins to the carla folder. I worked on Ubuntu. One more thing, how did you get the roadrunner?

**Reply by 15259400687** [reply]
![1](https://github.com/user-attachments/assets/09bd12ef-4a28-4216-98e9-be619a531590)
This is the plugin content under my CarlaUE4 \ plugin. The red box indicates the plugin that reported an error. My plugin was obtained from Mathwork
https://ww2.mathworks.cn/help/roadrunner/ug/downloading-plugins.html
However, when I used another method, which is to export FBX files to Carla, I did not need to use the plugin that reported the error mentioned above. However, when I followed the steps, I was able to import the scenes from Roadrunner into Carla, but no level files were generated. Is this normal? 
![2](https://github.com/user-attachments/assets/518315a5-8416-49a7-86cc-6d67c11397ee)
The above picture shows the content generated after importing FBX, but it does not generate a level type file like the manual import method。
Thank you again for your reply!

**Reply by 15259400687** [reply]
Sorry, I just realized that the level files it generates are stored in the Maps directory. Now I have a new question, is it because the screen of this newly generated level is black when running the Carla script, and there is no weather blueprint? Is there any way to make the weather of the level I imported look like the weather on Carla's built-in map?

**Comment by SExpert12** [comment]
Sorry buddy. No idea about weather. BTW, What you are trying to do in CARLA?

**Reply by 15259400687** [reply]
I wanna  use my custom scene map for traffic flow simulation experiments. But now we have been in the process of building our own map😭

**Comment by SExpert12** [comment]
Okay. Go ahead then you have built your map now. I don't think weather is a big deal when you are at this step.

Best of luck.
Welcome to CARLA. Lots of trial and error and depend on other people😂
